### ISSUE TITLE
I3010 False Positive for AWS::Logs::Transformer Properties.TransformerConfig

### ISSUE THREAD

**Issue by dothomson** [aws-cloudformation/cfn-lint#4684]
### CloudFormation Lint Version

1.56.0

### What operating system are you using?

Mac

### Describe the bug

<img width="511" height="259" alt="Image" src="https://github.com/user-attachments/assets/7658d45b-669e-40d3-b738-fdd4f4335d87" />

```txt
cfn-lint -c I3010 -t template.yaml

I3010 'Resources/ExampleTransformer/Properties/TransformerConfig/0' is approaching the limit of 1 properties
template.yaml:2449:11

I3010 'Resources/ExampleTransformer/Properties/TransformerConfig/1' is approaching the limit of 1 properties
template.yaml:2451:11
```

I have two elements in this array, the schema definition is: 

```json
{
 "TransformerConfig" : {
      "description" : "List of processors in a transformer",
      "type" : "array",
      "items" : {
        "$ref" : "#/definitions/Processor"
      },
      "minItems" : 1,
      "maxItems" : 20,
      "insertionOrder" : false
    }
}
```

I don't believe this should be flagged as an informational warning since i am in the items range. 

### Expected behavior

No output from cfn-lint

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"

Resources:
  ExampleTransformer:
    Type: AWS::Logs::Transformer
    Properties:
      LogGroupIdentifier: "exampleloggroup"
      TransformerConfig:
        - ParseJSON:
            Source: '@message'
        - AddKeys:
            Entries:
              - Key: function_name
                OverwriteIfExists: false
                Value: "function_name"
```
